athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/saphana/SAPHANAMetadataHandler.java

```diff
@@ -0,0 +1,196 @@
+/*-
+ * #%L
+ * athena-jdbc
+ * %%
+ * Copyright (C) 2019 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.connectors.athena.jdbc.saphana;
+
+import com.amazonaws.athena.connector.lambda.QueryStatusChecker;
+import com.amazonaws.athena.connector.lambda.data.Block;
+import com.amazonaws.athena.connector.lambda.data.BlockAllocator;
+import com.amazonaws.athena.connector.lambda.data.BlockWriter;
+import com.amazonaws.athena.connector.lambda.data.SchemaBuilder;
+import com.amazonaws.athena.connector.lambda.domain.Split;
+import com.amazonaws.athena.connector.lambda.domain.spill.SpillLocation;
+import com.amazonaws.athena.connector.lambda.metadata.GetSplitsRequest;
+import com.amazonaws.athena.connector.lambda.metadata.GetSplitsResponse;
+import com.amazonaws.athena.connector.lambda.metadata.GetTableLayoutRequest;
+import com.amazonaws.connectors.athena.jdbc.connection.DatabaseConnectionConfig;
+import com.amazonaws.connectors.athena.jdbc.connection.GenericJdbcConnectionFactory;
+import com.amazonaws.connectors.athena.jdbc.connection.JdbcConnectionFactory;
+import com.amazonaws.connectors.athena.jdbc.manager.JDBCUtil;
+import com.amazonaws.connectors.athena.jdbc.manager.JdbcMetadataHandler;
+import com.amazonaws.connectors.athena.jdbc.manager.PreparedStatementBuilder;
+import com.amazonaws.services.athena.AmazonAthena;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import org.apache.arrow.vector.complex.reader.FieldReader; 
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Handles metadata for SAPHANA. User must have access to `schemata`, `tables`, `columns`, `partitions` tables in
+ * information_schema.
+ */
+public class SAPHANAMetadataHandler
```

Comment:
naming convention

Reply:
SAP has multiple database products - SAP HANA, SAP AES, SAP MaxDB, SAP IQ and others. in future we may add support for other SAP databases as well. So i think this is required.